ui/src/routes/devices.$id.settings.macros.tsx

```diff
@@ -0,0 +1,1439 @@
+              </svg>
+            </div>
+            <div className="ml-3">
+              <h3 className="macro-error-text">{errorMessage}</h3>
+            </div>
+          </div>
+        </div>
+      )}
+
+      <div className="flex items-center justify-between mb-4">
+        <SettingsItem 
+          title="Macros"
+          description={`${macros.length}/${MAX_TOTAL_MACROS}`}
+        >
+          <div className="flex items-center gap-2">
+            {!showAddMacro && (
+              <Button
+                size="SM"
+                theme="primary"
+                text={isMaxMacrosReached ? `Max Reached` : "Add New Macro"}
+                onClick={() => setShowAddMacro(true)}
+                disabled={isMaxMacrosReached}
+              />
+            )}
+          </div>
+        </SettingsItem>
+      </div>
+
+      {loading && (
+        <div className="flex items-center justify-center p-8">
+          <LuLoader className="h-6 w-6 animate-spin text-blue-500" />
+        </div>
+      )}
+      <div className={`space-y-4 ${loading ? 'hidden' : ''}`}>
+        {showAddMacro && (
+          <div className="rounded-md border border-slate-200 bg-white p-3 dark:border-slate-700 dark:bg-slate-800">
+            <div className="flex items-center justify-between mb-2">
+              <h3 className="text-sm font-semibold text-black dark:text-white">Add New Macro</h3>
+            </div>
+            <div className="grid grid-cols-1 md:grid-cols-2 gap-2">
+              <div className="flex flex-col">
+                <input
+                  type="text"
+                  className={`macro-input ${errors.name ? 'border-red-500 dark:border-red-500' : ''}`}
+                  value={newMacro.name}
+                  onChange={e => {
+                    setNewMacro(prev => ({ ...prev, name: e.target.value }));
+                    if (errors.name) {
+                      const newErrors = { ...errors };
+                      delete newErrors.name;
+                      setErrors(newErrors);
+                    }
+                  }}
+                  placeholder="Macro Name"
+                />
+                <ErrorMessage error={errors.name} />
+              </div>
+              <div className="flex flex-col">
+                <input
+                  type="text"
+                  className={`macro-input ${errors.description ? 'border-red-500 dark:border-red-500' : ''}`}
+                  value={newMacro.description}
+                  onChange={e => {
+                    setNewMacro(prev => ({ ...prev, description: e.target.value }));
+                    if (errors.description) {
+                      const newErrors = { ...errors };
+                      delete newErrors.description;
+                      setErrors(newErrors);
+                    }
+                  }}
+                  placeholder="Description (optional)"
+                />
```

Comment:
I would suggest we remove the description - the title will be able to convey enough information to the consumer.

Reply:
good idea! removed description